packages/client/src/features/move-zoom/move-handler.ts

```diff
@@ -0,0 +1,161 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { Action, ChangeBoundsOperation, Point, SetViewportAction, Viewport } from '@eclipse-glsp/protocol';
+import { inject, injectable } from 'inversify';
+import { findParentByFeature, IActionDispatcher, IActionHandler, ICommand, isViewport, SModelRoot } from 'sprotty';
+import { EditorContextService } from '../../base/editor-context-service';
+import { TYPES } from '../../base/types';
+import { getElements, isSelectableAndBoundsAware, SelectableBoundsAware } from '../../utils/smodel-util';
+
+export interface MoveViewportAction extends Action {
+    kind: typeof MoveViewportAction.KIND;
+    moveX: number;
+    moveY: number;
+}
+
+export namespace MoveViewportAction {
+    export const KIND = 'moveViewportAction';
+
+    export function is(object: any): object is MoveViewportAction {
+        return Action.hasKind(object, KIND);
+    }
+
+    export function create(moveX: number, moveY: number): MoveViewportAction {
+        return { kind: KIND, moveX, moveY };
+    }
+}
+
+export interface MoveElementAction extends Action {
+    kind: typeof MoveElementAction.KIND;
+    elementIds: string[];
+    moveX: number;
+    moveY: number;
+}
+
+export namespace MoveElementAction {
+    export const KIND = 'moveElementAction';
+
+    export function is(object: any): object is MoveElementAction {
+        return Action.hasKind(object, KIND);
+    }
+
+    export function create(elementIds: string[], moveX: number, moveY: number): MoveElementAction {
+        return { kind: KIND, elementIds, moveX, moveY };
+    }
+}
+
+@injectable()
+export class MoveViewportHandler implements IActionHandler {
+    @inject(EditorContextService)
+    protected editorContextService: EditorContextService;
+
+    handle(action: Action): void | Action | ICommand {
+        if (MoveViewportAction.is(action)) {
+            const viewport = findParentByFeature(this.editorContextService.modelRoot, isViewport);
+            if (!viewport) {
+                return;
+            }
+
+            return this.moveViewport(viewport, action.moveX, action.moveY);
+        }
+    }
+
+    protected moveViewport(viewport: SModelRoot & Viewport, offsetX: number, offSetY: number): SetViewportAction {
+        const newViewport: Viewport = {
+            scroll: {
+                x: viewport.scroll.x + offsetX,
+                y: viewport.scroll.y + offSetY
+            },
+            zoom: viewport.zoom
+        };
+
+        return SetViewportAction.create(viewport.id, newViewport, { animate: true });
+    }
+}
+
+@injectable()
+export class MoveElementHandler implements IActionHandler {
+    @inject(EditorContextService)
+    protected editorContextService: EditorContextService;
+    @inject(TYPES.IActionDispatcher) protected dispatcher: IActionDispatcher;
+
+    handle(action: Action): void | Action | ICommand {
```

Comment:
Please also use the throttling of actions for the handlers in this file as well.